src/machine/machine_mimxrt1062_uart.go

```diff
@@ -0,0 +1,317 @@
+
+	// use 8 data bits, disable parity, use 1 idle char, and idle count starts
+	// after start bit
+	ctrlBits := uint32(((nxp.LPUART_CTRL_M_M_0 << nxp.LPUART_CTRL_M_Pos) & nxp.LPUART_CTRL_M_Msk) |
+		((nxp.LPUART_CTRL_PE_PE_0 << nxp.LPUART_CTRL_PE_Pos) & nxp.LPUART_CTRL_PE_Msk) |
+		((nxp.LPUART_CTRL_ILT_ILT_0 << nxp.LPUART_CTRL_ILT_Pos) & nxp.LPUART_CTRL_ILT_Msk) |
+		((nxp.LPUART_CTRL_IDLECFG_IDLECFG_0 << nxp.LPUART_CTRL_IDLECFG_Pos) & nxp.LPUART_CTRL_IDLECFG_Msk))
+	uart.Bus.CTRL.Set(ctrlBits)
+
+	rxSize, txSize := uart.getFIFOSize()
+
+	rxWater := rxSize >> 1
+	if rxWater > uint32(nxp.LPUART_FIFO_RXFIFOSIZE_Msk>>nxp.LPUART_FIFO_RXFIFOSIZE_Pos) {
+		rxWater = uint32(nxp.LPUART_FIFO_RXFIFOSIZE_Msk >> nxp.LPUART_FIFO_RXFIFOSIZE_Pos)
+	}
+
+	txWater := txSize >> 1
+	if txWater > uint32(nxp.LPUART_FIFO_TXFIFOSIZE_Msk>>nxp.LPUART_FIFO_TXFIFOSIZE_Pos) {
+		txWater = uint32(nxp.LPUART_FIFO_TXFIFOSIZE_Msk >> nxp.LPUART_FIFO_TXFIFOSIZE_Pos)
+	}
+
+	uart.Bus.WATER.Set(
+		((rxWater << nxp.LPUART_WATER_RXWATER_Pos) & nxp.LPUART_WATER_RXWATER_Msk) |
+			((txWater << nxp.LPUART_WATER_TXWATER_Pos) & nxp.LPUART_WATER_TXWATER_Msk))
+
+	// enable TX/RX FIFOs
+	uart.Bus.FIFO.SetBits(nxp.LPUART_FIFO_RXFE | nxp.LPUART_FIFO_TXFE)
+
+	// flush TX/RX FIFOs
+	uart.Bus.FIFO.SetBits(nxp.LPUART_FIFO_RXFLUSH | nxp.LPUART_FIFO_TXFLUSH)
+
+	uart.Bus.MODIR.SetBits( // set the CTS configuration/TX CTS source
+		((nxp.LPUART_MODIR_TXCTSC_TXCTSC_0 << nxp.LPUART_MODIR_TXCTSC_Pos) & nxp.LPUART_MODIR_TXCTSC_Msk) |
+			((nxp.LPUART_MODIR_TXCTSSRC_TXCTSSRC_0 << nxp.LPUART_MODIR_TXCTSSRC_Pos) & nxp.LPUART_MODIR_TXCTSSRC_Msk))
+
+	// clear all status flags
+	stat := uint32(nxp.LPUART_STAT_RXEDGIF_Msk | nxp.LPUART_STAT_IDLE_Msk | nxp.LPUART_STAT_OR_Msk |
+		nxp.LPUART_STAT_NF_Msk | nxp.LPUART_STAT_FE_Msk | nxp.LPUART_STAT_PF_Msk |
+		nxp.LPUART_STAT_LBKDIF_Msk | nxp.LPUART_STAT_MA1F_Msk | nxp.LPUART_STAT_MA2F_Msk)
+
+	// set data bits order
+	if uart.msbFirst {
+		stat |= nxp.LPUART_STAT_MSBF
+	} else {
+		stat &^= nxp.LPUART_STAT_MSBF
+	}
+
+	uart.Bus.STAT.SetBits(stat)
+
+	// enable RX/TX functions
+	uart.Bus.CTRL.SetBits(nxp.LPUART_CTRL_TE | nxp.LPUART_CTRL_RE)
+
+	// enable RX IRQ
+	uart.Interrupt.SetPriority(0xc0)
+	uart.Interrupt.Enable()
+
+	uart.configured = true
+}
+
+func (uart *UART) Disable() {
+
+	// first ensure the device is enabled
+	if uart.configured {
+
+		// wait for any buffered data to send
+		uart.Flush()
+
+		// stop trapping RX interrupts
+		uart.Interrupt.Disable()
+
+		// reset all internal registers
+		uart.resetTransmitting()
+
+		// disable RX/TX functions
+		uart.Bus.CTRL.ClearBits(nxp.LPUART_CTRL_TE | nxp.LPUART_CTRL_RE)
+
+		// put pins back into GPIO mode
+		uart.rx.Configure(PinConfig{Mode: PinInputPullUp})
+		uart.tx.Configure(PinConfig{Mode: PinInputPullUp})
+	}
+	uart.configured = false
+}
+
+// Flush blocks the calling goroutine until all data in the output buffer has
+// been written out.
+func (uart *UART) Flush() {
```

Comment:
Perhaps it would be a good idea to match `*os.File` here?

```suggestion
func (uart *UART) Sync() error {
```

Reply:
Changed `Flush()` to `Sync() error`, updated all references